lib/chibi/test.scm

```diff
@@ -797,21 +818,23 @@
            (number->string subgroups-run)
            (percent subgroups-pass subgroups-run))))
         (display (plural " subgroup" subgroups-pass))
-        (display " passed.\n")))))
-    (cond
-     ((test-group-ref group 'verbose)
-      (display
-       (test-header-line
-        (string-append "done testing " (or (test-group-name group) ""))
-        (or (test-group-indent-width group) 0)))
-      (newline)))
+        (display " passed.")))
+    (when (test-group-ref group 'verbose)
+      (display (test-group-line group #f)))
     (cond
      ((test-group-ref group 'parent)
       => (lambda (parent)
            (test-group-set! parent 'trailing #f)
            (test-group-inc! parent 'total-pass pass)
            (test-group-inc! parent 'total-fail fail)
-           (test-group-inc! parent 'total-error err))))))
+           (test-group-inc! parent 'total-error err)))
+     (else
+      (when (zero? (test-group-ref group 'level))
+        (newline))))))
+(define test-default-group-reporter
+  (case-lambda
```

Comment:
This shouldn't use case-lambda, and in fact if you are splitting the logic should use two separate parameters.

Reply:
Implemented